src/libraries/System.Diagnostics.Process/tests/ProcessTests.Unix.cs

```diff
@@ -171,14 +171,26 @@ public void ProcessNameMatchesScriptName()
                     string stat = File.ReadAllText($"/proc/{process.Id}/stat");
                     Assert.Contains($"({scriptName.Substring(0, 15)})", stat);
                     string cmdline = File.ReadAllText($"/proc/{process.Id}/cmdline");
-                    Assert.Equal($"/bin/sh\0{filename}\0", cmdline);
 
-                    Assert.Equal(scriptName, process.ProcessName);
+                    // cmdLine can sometimes be null: https://github.com/dotnet/runtime/issues/13757
+                    // according to docs, it's when the process is a zombie: https://man7.org/linux/man-pages/man5/proc.5.html
```

Comment:
The `Assert` that fails was added to understand better why `ProcessName` doesn't end up with the proper value. It verifies some expectations of the .NET implementation.

Can we verify the failing case has a zombie process? We can check `/stat` for the `Z` flag.

Reply:
Very good point, will do

Reply:
Thinking about it a bit more.

Probably this test fails when the `Assert` happens after the script has already exited. .NET will have reaped the process, and everything under `/proc` is gone.

This happens when sleep doesn't work, or when the asserts are delayed for 10 min. We can safely increase the 10 min timeout. And add an assert to check `!process.HasExited`.

Reply:
> This happens when sleep doesn't work

How would this be possible?

> when the asserts are delayed for 10 min

this is unlikely, as we just start the process and execute asserts right away

> everything under /proc is gone

Based on the CI logs it looks like not everything is gone. The previous line that asserts the content of `/proc/{process.Id}/stat` works just fine. Based on the available docs the only explanation I was able to find was the zombie process.